flict/flictlib/project/reader.py

```diff
@@ -0,0 +1,228 @@
+#!/usr/bin/env python3
+
+# SPDX-FileCopyrightText: 2022 Henrik Sandklef
+#
+# SPDX-License-Identifier: GPL-3.0-or-later
+
+import json
+import logging
+import os
+
+from flict.flictlib.return_codes import FlictError, ReturnCodes
+
+
+class Project:
+
+    def __init__(self, project_file, project_dirs):
+        project_reader = ProjectReaderFactory.get_projectreader(project_file, project_dirs)
+        self.project = project_reader.read_project()
+
+    @staticmethod
+    def dependencies_license(package):
+        licenses = []
+        for dep in package.get('dependencies', []):
+            #TODO: raise exception with nice message to user if license not present
+            licenses.append(f" ( {dep['license']}) ")
+        return " AND ".join(licenses)
+
+    @staticmethod
+    def package_license(package):
+        #TODO: raise exception with nice message to user if license not present
+        return package['license']
+
+    @staticmethod
+    def combined_work_license(package):
+        if Project.dependencies_license(package).strip() != "":
+            return f" ( {Project.package_license(package)} ) AND ( {Project.dependencies_license(package)} ) "
+        else:
+            return Project.package_license(package)
+
+
+class ProjectReader:
+    """Interface for classes that reads project from some kind of format"""
+    def read_project(self):
+        return
+
+    def _add_files_read(self, project_file):
+        self.files_read.append(project_file)
+
+    def _already_read(self, project_file):
+        return project_file in self.files_read
+
+    def _flatten_packages(self, packages):
+        package_list = []
+
+        for pkg in packages.items():
```

Comment:
a better pattern for a dict is `for k, v in packages.items()` then you can use `k` (for the key) and `v` for the value.
That looks much more readable than the tuple with an index handling below

Reply:
:)